### DISCUSSION TITLE
Identify Neighbors Uniquely

### DISCUSSION THREAD

**Discussion by ssenthu** [openthread/openthread#9481] [Q&A] [answered]
Is that possible to get the eui64 of neighbors? extmac address seems changing over time. How we can uniquily identify the neighbors?

**Comment by jwhui** [comment]
Thread intentionally avoids using static identifiers, such as IEEE EUI-64. Thread requires devices to randomly generate the Extended Address after factory reset.

The application layer should be responsible for maintaining and reporting any identifiers that are necessary to support the application.

**Reply by ssenthu** [reply]
Thanks @jwhui . 
1) Is there possibility to have same ExtAddr to two different routers when randomly generated?  or is that generation  specific to EUI64?

2) Only FactoryReset will regenerate the ExtAddr?  Or simple reset or changeing the dataset parameterers will create new ExtAddr too?

**Reply by jwhui** [reply, marked as answer]
> Is there possibility to have same ExtAddr to two different routers when randomly generated? or is that generation specific to EUI64?

The likelihood of collisions on a 64-bit random value is exceedingly low.

> Only FactoryReset will regenerate the ExtAddr? Or simple reset or changeing the dataset parameterers will create new ExtAddr too?

Only factory reset will regenerate the Extended Address.

**Reply by ssenthu** [reply]
Thanks a lot Jonathan @jwhui. You are extremely helpful.
